lib/control/include/clingo/control/solver.hh

```diff
@@ -615,8 +686,11 @@ class Solver : public BaseView {
     void add_const(String name, Symbol value);
     //! Get the const map.
     [[nodiscard]] auto const_map() -> Input::ConstMap const &;
+    //! Ground the program asynchronously.
+    [[nodiscard]] auto start_ground(ProgramParamVec params, UGroundEventHandler handler) -> GroundHandle;
     //! Ground the program.
-    void ground(ProgramParamVec const &params, Ground::ScriptCallback *ctx);
+    auto ground(ProgramParamVec const &params, Ground::ScriptCallback *ctx, Util::StopFlag *stop = nullptr)
```

Comment:
Do you think it makes sense to check for `stop->stop_requested()` on entry to this function or somewhere else early in the grounding process?